### PR TITLE
[FIX/IMP] API method set_partner_id_from_partner_address_id switched to SQL

### PR DESCRIPTION
**Pull request by pedrobaeza**
Same as #236

### CODE REVIEW

**Comment by legalsylvain** [comment]
Thanks.

**Comment by hbrunn** [comment]
:+1:

**Comment by StefanRijnhart** [comment]
Please make it a single query. I think you can use UPDATE...FROM.

**Comment by pedrobaeza** [comment]
I haven't touched that part for this PR.

**Comment by StefanRijnhart** [comment]
No, but it makes no sense to loop over the results of an sql join to do an sql update. You should make it a single query.

**Comment by pedrobaeza** [comment]
@StefanRijnhart , I have joined both queries. Please review it.

**Comment by pedrobaeza** [comment]
Used the same SQL succesfully on 7.0 (where it applies).

**Comment by pedrobaeza** [comment]
Can anyone review this?

File: openerp/openupgrade/openupgrade_70.py [hunk 1]
```diff
@@ -42,14 +41,18 @@ def set_partner_id_from_partner_address_id(
...
+        "UPDATE %s "
+        "SET %s=src.partner_id "
+        "FROM ("
+        "    SELECT target.id as id, "
+        "           address.openupgrade_7_migrated_to_partner_id as partner_id"
+        "    FROM %s as target, res_partner_address as address "
+        "    WHERE address.id = target.%s"
+        ") src "
+        "WHERE id=src.id" %
+        (table, partner_field, table, address_field))
```
Comment (StefanRijnhart) [inline openerp/openupgrade/openupgrade_70.py]: You can eliminate the subquery using

```
UPDATE %{table}s
SET %{partner_field}s=address.openupgrade_7_migrated_to_partner_id
FROM res_partner_address address
WHERE %{table}s.%{address_field}=address.id
```

**Comment by pedrobaeza** [comment]
@StefanRijnhart, thanks for the tip. UPDATE syntax is very confusing for me.

When you validate this, I'll apply it to 7.0 also.

**Comment by StefanRijnhart** [comment]
:+1:, thanks!

**Pull request merged**